### DISCUSSION TITLE
Possible to export video from ruffle?

### DISCUSSION THREAD

**Discussion by timmerk** [ruffle-rs/ruffle#3360] [Q&A]
I know there is the exporter, but it seems to just export frames. Any way to export the entire swf as it plays in ruffle as a video?

**Comment by danielhjacobs** [comment]
Not that I know of within Ruffle, but rather than exporting it as it plays in Ruffle, this application was created by the same person who first created Ruffle to convert Adobe Flash videos to movies: https://www.newgrounds.com/wiki/creator-resources/flash-resources/swivel?path=/wiki/creator-resources/flash-resources/swivel

**Reply by timmerk** [reply]
Thank you! I've tried that, but it freezes on both the Mac version and Windows version when I load a simple SWF that plays fine in Ruffle. Maybe I'll try building the latest from source.

**Comment by torokati44** [comment]
Ruffle can't export video directly at the moment, but once you have the frames as a sequence of images, there are a lot of options to encode them into any kind of video!
I'm not sure which one would be the easiest for you, but at least HandBrake, VLC, Blender, and even `ffmpeg` (from the command line) can do that for you.

**Reply by WindowsNT** [reply]
How can one get the frames?

**Reply by tsunamistate** [reply]
Clone ruffle and execute this in its root folder:
`cargo run --release --package=exporter -- path/to/file.swf path/to/screenshots --frames [number of frames in swf]`

**Reply by WindowsNT** [reply]
Hmm. I don't see any exporting function in ruffle.exe --help, is this feature only available in linux?

**Reply by tsunamistate** [reply]
Please read the README file, specifically, "Building from source" and "Exporter" sections.
Exporter is a separate executable that is not packaged into releases at the moment

**Reply by moltenform** [reply]
@tsunamistate  Ran this yesterday and it worked perfectly, thank you!
